### ISSUE TITLE
Vue 3 support

### ISSUE THREAD

**Issue by mabarbeau** [ankurk91/vue-flatpickr-component#198]
**I'm submitting a ...**  (check one with "x")
```
[ ] Bug report => search github for a similar issue or PR before submitting
[x] Feature request
[ ] Other, please describe
```

**Tell about your platform**
* flatPickr version : 4.6.6
* Vue.js version : 3.0.0-0
* Browser name and version : Chrome 85.0.4183.83 (Official Build) (64-bit)
* This package version : 8.1.6

**Current behavior**
```
runtime-core.esm-bundler.js?5c40:38 [Vue warn]: Unhandled error during execution of render function 
  at <FlatPickr modelValue=null onUpdate:modelValue=fn > 
  at <App>
warn @ runtime-core.esm-bundler.js?5c40:38
```
```
runtime-core.esm-bundler.js?5c40:216 Uncaught TypeError: t is not a function
    at Proxy.render (vue-flatpickr.min.js?c38f:1)
    at renderComponentRoot (runtime-core.esm-bundler.js?5c40:535)
    at componentEffect (runtime-core.esm-bundler.js?5c40:4286)
    at reactiveEffect (reactivity.esm-bundler.js?a1e9:42)
    at effect (reactivity.esm-bundler.js?a1e9:17)
    at setupRenderEffect (runtime-core.esm-bundler.js?5c40:4269)
    at mountComponent (runtime-core.esm-bundler.js?5c40:4227)
    at processComponent (runtime-core.esm-bundler.js?5c40:4183)
    at patch (runtime-core.esm-bundler.js?5c40:3788)
    at mountChildren (runtime-core.esm-bundler.js?5c40:3996)
```
**Expected behavior**
Works the same as when using vue 2

**Minimal reproduction of the problem with instructions**

See: https://github.com/mabarbeau/demo-vue-3-flatpickr-component

OR 

1. Update `vue-cli`
1. `vue create demo-vue-3-flatpickr-component`
1. Select "Default (Vue 3 Preview) ([Vue 3] babel, eslint)"
1.  Create a componet using the [minimal example](https://github.com/ankurk91/vue-flatpickr-component#minimal-example)

**Comment by ankurk91** [comment]
Vue 3 has not been released yet, i will look into once first stable version is out.

**Comment by ankurk91** [comment]
I have made some progress,

Looks like it is an easy update
https://codesandbox.io/s/flatpickr-in-vuejs-v3-80151

Still have to look into test cases.

**Comment by jd-0001** [comment]
+1

**Comment by sudip571** [comment]
hi
when will it available for vue 3?
vue 3 has already been released on sept 18th, I guess

**Comment by ankurk91** [comment]
No everything is ready for vue 3, for example the test util.

Though i have made progress, it still need manual testing

**Comment by sudip571** [comment]
will it be compatible with composition API and TypeScript?

**Comment by ankurk91** [comment]
@sudip571 

You can see the progress on "next" branch, i have no plans to write this into TS for now.

**Comment by ankurk91** [comment]
Just released v9.0.0

You can try this version like:
```
npm install vue-flatpickr-component@next
```

Please try and send feedbacks

**Comment by TixieBorg** [comment]
Hi, thanks for working on a vue3 already!
I saw you made some changes between your last codesandbox link and the v9.0.0 release, and looks like it's completely unusable with the `allowInput` option activated: [https://codesandbox.io/s/flatpickr-in-vuejs-v3-forked-1ku25](https://codesandbox.io/s/flatpickr-in-vuejs-v3-forked-1ku25?file=/src/Datepicker.vue) (for comparaison the same example with the old codesandbox code version: [https://codesandbox.io/s/flatpickr-in-vuejs-v3-forked-s43hn](https://codesandbox.io/s/flatpickr-in-vuejs-v3-forked-s43hn?file=/src/Datepicker.vue)).

**Comment by ankurk91** [comment]
I am aware of `allowInput` option issue.
I could not figure out why it is no longer allowing user to type manually. 

Any help would be appreciated.

**Comment by ankurk91** [comment]
I have fixed the `allowInput` issue, please try again the new version
```
npm install vue-flatpickr-component@next
```

**Comment by TixieBorg** [comment]
Looks like it works [just fine](https://codesandbox.io/s/flatpickr-in-vuejs-v3-forked-0viv6?file=/src/Datepicker.vue) 👌

**Comment by ankurk91** [comment]
Released v9.0.2 , stable
